src/lava/proc/monitor/process.py

```diff
@@ -0,0 +1,254 @@
+    RefPorts and Vars are created dynamically, as the Monitor process is
+    used to probe OutPorts and Vars of other processes. For this purpose,
+    Monitor process has probe(..) function, which as arguments takes the
+    target Var or OutPorts and number of time steps we want to monitor given
+    process.
+    multiple instance variables
+
+    Attributes
+    ----------
+    data : dict
+        Dictionary that is populated by monitoring data once get_data(..)
+        method is called, has the follwoing structure:
+        data
+          \__Monitored_process_name
+             \__Monitored_var_or_outPort_name
+
+    proc_params: dict
+        Process parameters that will be transferred to the corresponding
+        ProcessModel. It is populated with the names of dynamically
+        created port and var names of Monitor process, to be carried to its
+        ProcessModel. It is a dictionary of the following structure:
+            "RefPorts": names of RefPorts created to monitor target Vars
+            "VarsRead": names of Vars created to store data from target Vars
+            "InPorts": names of InPorts created to monitor target OutPorts
+            "OutsRead": names of Vars created to store data from target OutPorts
+            "n_ref_ports": number of created RefPorts, also monitored Vars
+            "n_in_ports": number of created InPorts, also monitored OutPorts
+
+    target_names: dict
+        The names of the targeted Processes and Vars/OutPorts to be monitored.
+        This is used in get_data(..) method to access the target names
+        corresponding to data-storing Vars of Monitor process during readout
+        phase. This dict has the follwoing sturcture:
+            key: name of the data-storing Vars, i.e. VarsRead and OutsRead
+            value: [Monitored_process_name, Monitored_var_or_outPort_name]
+
+    Methods
+    -------
+    post_init()
+        Create one prototypical RefPort, InPort and two Vars. This ensure
+        coherence and one-to-one correspondence between Monitor process and
+        ProcessModel in terms LavaPyTypes and Ports/Vars. These prototypical
+        ports can later be updated inside probe(..) method.
+
+    probe(target, num_steps)
+        Probe the given target for num_step time steps, where target can be
+        a Var or OutPort of some process.
+
+    get_data()
+        Fetch the monitoring data from the Vars of Monitor process that
+        collected it during the run from probed process, puts into dict form
+        for easier access by user
+    """
+    def __init__(self, **kwargs):
+        """
+        Initialize the attributes and run post()
+        """
+        super().__init__(**kwargs)
+        # self.ref = []
+        self.data = {}
+
+        self.proc_params["RefPorts"] = []
+        self.proc_params["VarsRead"] = []
+        self.proc_params["InPorts"] = []
+        self.proc_params["OutsRead"] = []
+        self.proc_params["n_ref_ports"] = 0
```

Comment:
For now it is necessary, as we create both a RefPort and an InPort, regardless, if we are gonna use them. This itself is necessary for now to be sure that Process and ProcessModel match, and to be able to monitor both Vars and OutPorts

Reply:
I do not understand. Is self.proc_params["n_ref_ports"] == len(self.proc_params["RefPorts"] not always True?

Reply:
That is always true, but I want to differentiate between dummy a RefPort and one that is actually used (those that are connected to Var). This will not be necessary when we solve multi-var probing as I said